packages/entity-server/src/routes/Route.ts

```diff
@@ -34,7 +36,7 @@ export class Route {
     }
   }
 
-  async buildResponse(): Promise<Record<string, unknown>> {
+  async buildResponse(): Promise<ResBody<Res>> {
```

Comment:
This is pretty cool, it means that Typescript can infer the return type of this function based on the Response type for each route, now we get proper errors if we return an unexpected response object 👍

Reply:
Nice, that is quite cool. I must say the Typescript syntax is taking some getting used to though! It's powerful, but that power comes through abstractions that I still need to learn

Reply:
Yeah I feel a bit of a similar way, most of the syntax is pretty unfamiliar to me and it's a bit daunting. The plus side is that when you finally get it right, vscode plays so nicely with it that I find it really re-assuring 😄